src/core/socialprovider.js

```diff
@@ -477,12 +478,64 @@ XMPPSocialProvider.prototype.onOnline = function(continuation) {
   this.vCardStore.updateProperty(this.id, 'status', 'ONLINE');
   this.vCardStore.refreshContact(this.id, null);
 
+  this.pollForDisconnect_();
+
   continuation(this.vCardStore.getClient(this.id));
 };
 
+XMPPSocialProvider.prototype.ping_ = function() {
+  var pingDate = new Date();
+  var ping = new window.XMPP.Element('iq', {type: 'get'})
+      .c('ping', {'xmlns': 'urn:xmpp:ping'}).up();
+  this.client.send(ping);
+
+  // Check that we got a response from the server after the ping was sent.
+  setTimeout(function() {
+    if (this.client &&  // check this.client to be sure logout wasn't called.
+        (!this.lastMessageDate_ || this.lastMessageDate_ < pingDate)) {
+      // No response to ping, we are disconnected.
+      console.warn('No ping response from server, logging out');
+      this.logout();
+    }
+  }.bind(this), 10000);
+};
+
+XMPPSocialProvider.prototype.pollForDisconnect_ = function() {
+  if (this.pollForDisconnectInterval_) {
+    console.error('pollForDisconnect_ called while already polling');
```

Comment:
This should never get triggered, it's actually just a sanity check to make sure we don't end up calling setInterval twice and doubling the amount of callbacks/pings we make